server/server.js

```diff
@@ -22,6 +23,37 @@ app.use(express.static(reactStaticDir));
 app.use(express.static(uploadsStaticDir));
 app.use(express.json());
 
+// creating a new user
+app.post('/api/users', async (req, res, next) => {
+  const { username, password } = req.body;
+  try {
+    const query = 'INSERT INTO "Users" (username, password) VALUES ($1, $2)';
+    await db.query(query, [username, password]);
+    res.status(201).json({ message: 'User created successfully' });
+  } catch (error) {
+    next(error);
+  }
+});
+
+// login
+app.post('/api/login', async (req, res, next) => {
+  const { username, password } = req.body;
+  try {
+    const user = await db.query('SELECT * FROM "Users" WHERE username = $1', [username]);
+    if (!user) {
+      return res.status(401).json({ message: 'Invalid username or password' });
```

Comment:
In our express-error-handling exercise we used error handling middleware and threw an exception here. You should do the same for the final project.